molo/forms/utils.py

```diff
@@ -28,20 +28,32 @@ def __init__(self, object_list, data=dict(), answered=dict()):
         ]
 
         self.page_breaks = []
-        for i, field in enumerate(self.object_list):
-            # Remove page breaks before questions that have been skipped
+        i = 0
+        while i < len(self.object_list):
+            field = self.object_list[i]
+            i += 1
+
+            # Don't add breaks for unsaved or hidden fields
+            if (not field.pk) or field.field_type == 'hidden':
+                continue
```

Comment:
if field.pk and field.field_type != 'hidden':
                ...rest for the logic block
Id go for something a lil like this

Reply:
I was aiming to avoid more nesting for the sake of readability